persistence/src/main/java/com/epam/esm/persistence/impl/CertificateDAOImpl.java

```diff
@@ -0,0 +1,109 @@
+            "select * from module2_certificates_tags.certificates " +
+                    "where name like '%";
+    private static final String SECOND_PART_OF_SQL_QUERY_READ_CERTIFICATES_BY_PART_OF_NAME_AND_DESCRIPTION = "%' or description like '%";
+    private static final String THIRD_PART_OF_SQL_QUERY_READ_CERTIFICATES_BY_PART_OF_NAME_AND_DESCRIPTION = "%'";
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public CertificateDAOImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public List<Certificate> allCertificates() {
+        return jdbcTemplate.query(SQL_QUERY_READ_CERTIFICATES_LIST, new BeanPropertyRowMapper<>(Certificate.class));
+    }
+
+    @Nullable
+    @Override
+    public Certificate certificateById(long id) {
+        return jdbcTemplate.query(SQL_QUERY_READ_ONE_CERTIFICATE_BY_ID,
+                new BeanPropertyRowMapper<>(Certificate.class), id)
+                .stream().findAny().orElse(null);
+    }
+
+    @Override
+    public void update(long id, Certificate certificate) {
+        jdbcTemplate.update(SQL_QUERY_UPDATE_CERTIFICATE, certificate.getName(),
+                certificate.getDescription(), certificate.getPrice(), certificate.getDuration(),
+                certificate.getCreateDate(), certificate.getUpdateLastDate(), id);
+    }
+
+    @Override
+    public long create(Certificate certificate) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection
+                    .prepareStatement(SQL_QUERY_INSERT_CERTIFICATE, Statement.RETURN_GENERATED_KEYS);
+            ps.setString(1, certificate.getName());
+            ps.setString(2, certificate.getDescription());
+            ps.setObject(3, certificate.getPrice());
+            ps.setInt(4, certificate.getDuration());
+            ps.setObject(5, certificate.getCreateDate().getEpochSecond());
+            ps.setObject(6, certificate.getUpdateLastDate().getEpochSecond());
+            return ps;
+        }, keyHolder);
+        return keyHolder.getKey().intValue();
+    }
+
+    @Override
+    public void delete(long id) {
+        jdbcTemplate.update(SQL_QUERY_DELETE_CERTIFICATE_BY_ID, id);
+    }
+
+    @Override
+    public List<Certificate> getCertificatesByTagId(long tagId) {
+        return jdbcTemplate.query(SQL_QUERY_READ_CERTIFICATE_LIST_BY_TAG_ID
+                , new BeanPropertyRowMapper<>(Certificate.class), tagId);
+    }
+
+    @Override
+    public void addCertificateTag(long certificateId, long tagId) {
+        jdbcTemplate.update(SQL_QUERY_INSERT_ID_CERTIFICATE_AND_ID_TAG, certificateId, tagId);
+    }
+
+    @Override
+    public List<Certificate> getCertificatesByPartOfNameOrDescription(String name, String description) {
+        return jdbcTemplate.query(FIRST_PART_OF_SQL_QUERY_READ_CERTIFICATES_BY_PART_OF_NAME_AND_DESCRIPTION + name +
```

Comment:
reimplement this method to search by one term in both fields and don't forget to **always** use prepared statements! there is no difference if it's a **like** operator or any other